Fix SIMD loop stop condition for length=0

Constellation opened the pull request:
#### 4528a8ead9b4d7cc80e92f7f969e3ae674c2bd57
<pre>
Fix SIMD loop stop condition for length=0
<a href="https://bugs.webkit.org/show_bug.cgi?id=245017">https://bugs.webkit.org/show_bug.cgi?id=245017</a>
&lt;rdar://99768369&gt;

Reviewed by Mark Lam.

We should stop the loop when the remaining length becomes zero.
Tested via guarded malloc bots.

* Source/WTF/wtf/text/StringCommon.cpp:
(WTF::find16AlignedImpl):
(WTF::find32AlignedImpl):
(WTF::find64AlignedImpl):
(WTF::findFloatAlignedImpl):
(WTF::findDoubleAlignedImpl):
(WTF::find8NonASCIIAlignedImpl):
(WTF::find16NonASCIIAlignedImpl):

Canonical link: <a href="https://commits.webkit.org/254349@main">https://commits.webkit.org/254349@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5fcaff969685a1b08a7605a112c75ee3944bd67f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/88791 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/33357 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/19676 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/97995 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/154502 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/31862 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/27473 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/81036 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/92616 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/94421 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/25278 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/75777 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/25231 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/80153 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/80198 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/68193 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/80513 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/29660 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/14216 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/74294 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/29390 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/15213 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/26272 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3057 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/32827 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/38139 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/77156 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/31515 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34320 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/17103 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

MenloDorian commented on Source/WTF/wtf/text/StringCommon.cpp:
```diff
@@ -83,7 +83,7 @@ const uint16_t* find16AlignedImpl(const uint16_t* pointer, uint16_t character, s
         length -= stride;
         cursor += stride;
-    } while (length < oldLength);
+    } while (length && length < oldLength);
```
It looks like the purpose of the `length < oldLength` condition is to detect an underflow.  The only way that is possible is if `length` is less than stride.  Why not change this condition to `length >= stride`?

MenloDorian replied:
We can also remove `oldLength` since it is not needed anymore.

Constellation replied:
Fixed.

MenloDorian reviewed (approved):
r=me

Constellation committed:
Fix SIMD loop stop condition for length=0
https://bugs.webkit.org/show_bug.cgi?id=245017
<rdar://99768369>

Reviewed by Mark Lam.

We should stop the loop when the remaining length becomes zero.
Tested via guarded malloc bots.

* Source/WTF/wtf/text/StringCommon.cpp:
(WTF::find16AlignedImpl):
(WTF::find32AlignedImpl):
(WTF::find64AlignedImpl):
(WTF::findFloatAlignedImpl):
(WTF::findDoubleAlignedImpl):
(WTF::find8NonASCIIAlignedImpl):
(WTF::find16NonASCIIAlignedImpl):

Canonical link: https://commits.webkit.org/254349@main
